crates/re_data_store/src/entity_tree.rs

```diff
@@ -88,22 +88,47 @@ pub struct SubtreeInfo {
     ///
     /// ⚠ Auto-generated instance keys are _not_ accounted for. ⚠
     pub time_histogram: TimeHistogramPerTimeline,
+
+    /// Number of bytes used by all arrow data in this tree (ignores overhead from book-keeping, schemas, etc).
+    data_bytes: u64,
 }
 
 impl SubtreeInfo {
     /// Assumes the event has been filtered to be part of this subtree.
     fn on_event(&mut self, event: &StoreEvent) {
+        use re_types_core::SizeBytes as _;
+
         match event.kind {
             StoreDiffKind::Addition => {
                 self.time_histogram
                     .add(&event.times, event.num_components() as _);
+
+                for cell in event.cells.values() {
+                    self.data_bytes += cell.heap_size_bytes();
```

Comment:
The cells themselves are also stored on the heap at this point, so you want `total_size_bytes()`